File: app/src/main/java/com/udbstudents/tseapp/adapters/MyAdapter.java
package com.udbstudents.tseapp.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    private Context mCtx;
    private List<Category> categoryList;

    public MyAdapter(Context mCtx, List<Category> categoryList){
        this.mCtx = mCtx;
        this.categoryList = categoryList;
    }
    @NonNull
    @Override
    public MyAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(mCtx);
        View view = inflater.inflate(R.layout.item_main, null);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyAdapter.MyViewHolder holder, int position) {

        //getting the product of the specified position
        Category category = categoryList.get(position);

        //binding the data with the viewholder views
        holder.textViewName.setText(category.getName());
        holder.imageView.setImageDrawable(mCtx.getResources().getDrawable(category.getImage()));

    }

    @Override
    public int getItemCount() {
        return categoryList.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {

        TextView textViewName;
        ImageView imageView;

        public MyViewHolder(View itemView) {
            super(itemView);

            textViewName = itemView.findViewById(R.id.categoryName);
            imageView = itemView.findViewById(R.id.categoryImage);
        }
    }


    /*
    private String[] mDataset;

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView textView;
        public MyViewHolder(TextView v) {
            super(v);
            textView = v;
        }
    }

    // Provide a suitable constructor (depends on the kind of dataset)
    public MyAdapter(String[] myDataset) {
        mDataset = myDataset;
    }

    // Create new views (invoked by the layout manager)
    @Override
    public MyAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent,
                                                     int viewType) {
        // create a new view
        TextView v = (TextView) LayoutInflater.from(parent.getContext()).inflate(R.layout.item_main, parent, false);
        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        holder.textView.setText(mDataset[position]);

    }

    // Return the size of your dataset (invoked by the layout manager)
    @Override
    public int getItemCount() {
        return mDataset.length;
    }*/
}

File: app/src/main/java/com/udbstudents/tseapp/adapters/MyAdapter.kt
package com.udbstudents.tseapp.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Filter
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.udbstudents.tseapp.R
import com.udbstudents.tseapp.models.ActaAndMunicipio
import java.util.*
import kotlin.collections.ArrayList

class MyAdapter(private val events: MutableList<ActaAndMunicipio>) :





    RecyclerView.Adapter<MyAdapter.ViewHolder>() {

    private var emptyList : MutableList<ActaAndMunicipio> = mutableListOf()
    private var emptylistFull: MutableList<ActaAndMunicipio> = mutableListOf()


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyAdapter.ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_event, parent, false)
        )
    }

    override fun getItemCount(): Int {
        return events.size
    }

    fun getFilter(): Filter? {
        return exampleFilter
    }

    private val exampleFilter: Filter = object : Filter() {
        override fun performFiltering(constraint: CharSequence): FilterResults {
            val filteredList: MutableList<ActaAndMunicipio> = ArrayList()
            if (constraint == null || constraint.length == 0) {
                filteredList.addAll(emptylistFull)
            } else {
                val filterPattern =
                    constraint.toString().toLowerCase(Locale.ROOT).trim { it <= ' ' }
                for (item in emptylistFull) {
                    if (item.municipio.nombre!!.toLowerCase(Locale.ROOT).contains(filterPattern)) {
                        filteredList.add(item)
                    }
                }
            }
            val results = FilterResults()
            results.values = filteredList
            return results
        }

        override fun publishResults(
            constraint: CharSequence,
            results: FilterResults
        ) {

            events.clear()
            events.addAll(results.values as MutableList<ActaAndMunicipio>)
            notifyDataSetChanged()
        }
    }
    override fun onBindViewHolder(holder: MyAdapter.ViewHolder, position: Int) {
        val event = events[position]

        holder.actaIndice.text =event.actaIndece.toString()
        holder.municipioName.text = event.municipio.nombre
        holder.partidoUno.text = event.acta.idPartido_uno
        holder.partidodos.text = event.acta.idPartido_dos
        holder.partidotres.text = event.acta.idPartido_tres
        holder.partidocuatro.text = event.acta.idPartido_cuatro
        holder.votosUno.text = event.acta.votos_partUno.toString()
        holder.votosDos.text = event.acta.votos_partDos.toString()
        holder.votosTres.text = event.acta.votos_partTres.toString()
        holder.votosCuatro.text = event.acta.votos_partCuatro.toString()
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val actaIndice = view.findViewById<TextView>(R.id.event_index_txt)
        val municipioName = view.findViewById<TextView>(R.id.municipioName)
        val partidoUno = view.findViewById<TextView>(R.id.partidounoitem)
        val partidodos = view.findViewById<TextView>(R.id.partidoDositem)
        val partidotres = view.findViewById<TextView>(R.id.partidoTresitem)
        val partidocuatro = view.findViewById<TextView>(R.id.partidoCuatroitem)
        val votosUno = view.findViewById<TextView>(R.id.partidounoName)
        val votosDos = view.findViewById<TextView>(R.id.partidoDosName)
        val votosTres = view.findViewById<TextView>(R.id.partidoTresName)
        val votosCuatro = view.findViewById<TextView>(R.id.partidoCuatroName)
    }
}